src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -325,10 +326,62 @@ private SolutionState AddProject(ProjectState projectState)
             dependencyGraph: newDependencyGraph);
     }
 
+    private SolutionState AddProjects(ArrayBuilder<ProjectState> projectStates)
+    {
+        if (projectStates.Count == 0)
+            return this;
+
+        if (projectStates.Count == 1)
+            return AddProject(projectStates.First());
```

Comment:
>return AddProject(projectStates.First()); [](http://example.com/codeflow?start=12&length=41)

Any reason to not to reverse this and just have AddProject call into AddProjects?

Reply:
You tackled this a bit different that I did when I tried this and I'm curious about that. Essentially, I just made AddProjects the same as AddProject with a loop around it's contents.  (ie, I kept the dependency graph buildup the same). Feels like the depGraph being updated like this would be preferable.

Reply:
done.